src/components/battles/animations/shared/Fainting.ts

```diff
@@ -45,6 +45,19 @@ export class Fainting<TGameStartr extends FullScreenPokemon> extends GeneralComp
                 const playerName = this.gameStarter.itemsHolder.getItem("name");
                 const partyIsWipedText: (string | string[])[][] = [[pokemon.nickname, " fainted!"]];
 
+                if (teamName === "opponent") {
```

Comment:
The code in this block should be separated into other functions in other class(es). We'll need to call most of it in other places: for example, when Exp All / Exp Share are added.

Also, there'll need to be some way to add a dialog for teaching the Pokemon any new moves it can now learn.